### ISSUE TITLE
The OAuth2 extension should be able to set SSLContext with a server certificate in the native mode

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#3435]
**Description**
The OAuth2 extension makes an HTTPS call to the token introspection endpoint requiring a server certificate. The only option right now in the native mode is to update the global java key store. The user application should be able to ship public certificates only without having to depend on the admin work be done first on a given host   

**Implementation ideas**
Investigate how to fix it with GraalVM, possibly open a GraalVM PR if needed

**Comment by sberyozkin** [comment]
@loicmathieu FYI.

**Comment by loicmathieu** [comment]
@sberyozkin this is on my todo list, it's possible for hotspot with a non-documented configuration property, but the native part still needs to be done.

**Comment by sberyozkin** [comment]
@loicmathieu Hi Loic, I've only CC-ed FYI :-). if you can find some cycles to investigate then it would be good, or perhaps we can collaborate on this one later on. Surely we'll sort this issue out eventually :-). And thanks for getting it highlighted with your PR in the 1st place, cheers

**Comment by loicmathieu** [comment]
@sberyozkin this will need to be investiguated by a GraalVM expert as the current code to include a server certificate contains a call to `SSLContext.getInstance()` and the current SSLContext substitution that Quarkus provides only contains substitution for `SSLContext.getDefault()`

I enabled SSL for the extension using a `ExtensionSslNativeSupportBuildItem` but it didn't works.

I don't know who can help me on the GraalVM part from the Quarkus team ?

**Comment by loicmathieu** [comment]
@sberyozkin I create the PR #3456 that enables SSL in native builds and document the already existing `quarkus.oauth2.ca-cert-file` property.

I tested a lot of different way of trying to make this property works on native mode (including playing with the substitution of SSLContext!) but I ended up in the conclusion that this is not possible. SSLContext cannot be redefined in GraalVM in the Quarkus context for now, so the server ca cert file needs to be included in the trust store of GraalVM as explained in the SSL native guide.

As I said, maybe someone with more knowledge of SSL native support with Quarkus may find a way to achieve this but for now the PR is the best I can do.
